check_no_globals.go

```diff
@@ -18,6 +18,62 @@ type allowedExpression struct {
 	SelName string
 }
 
+// reports is the reports of found global variables that's not valid and will be
+// reported to the analysis pass.
+type report struct {
+	name string
+	pos  token.Pos
+}
+
+// Analyzer is the analasys analyzer for gochecknoglobals. Ironically enough,
+// this is in fact a global variable.
+var Analyzer = &analysis.Analyzer{ //nolint
+	Name:             "gochecknoglobals",
+	Doc:              "Don't allow global variables",
+	Run:              run,
+	Flags:            flags(),
+	RunDespiteErrors: true,
+}
```

Comment:
I think moving `Analyzer` from being a package var to being a func would be a good move. Is there a reason that wouldn't work?
```suggestion
func Analyzer() *analysis.Analyzer {
	return &analysis.Analyzer{
		Name:             "gochecknoglobals",
		Doc:              "Don't allow global variables",
		Run:              run,
		Flags:            flags(),
		RunDespiteErrors: true,
	}
}
```

Reply:
Nope this will work, just not the most common way to do it. But let's do it here, this is `gochecknoglobals` after all :D